feat: adds fundamentals 4 introduction (#25983)
TheOdinProject/curriculum: foundations/javascript_basics/fundamentals-4.md

### Introduction

> One of the challenges of programming is dealing with large amounts of data. For example, if you want to store the names of all the students in your class, how would you do it? You could create a variable for each name, but that would be tedious and inefficient, not to mention hard to manage and update. For example, what if you want to change or access the names later?

Reviewer: [Proofreading] This paragraph is off to a great start. It just needed a bit of rewording to make it clear. :) 
```suggestion
One of the challenges of programming is dealing with large amounts of data. For example, if you want to store the names of all the students in your class, how would you do it? You could create a variable for each name, but that would be tedious and inefficient. It'll also be hard to manage and update. What if you want to change or access the names later?
```

Author: Good changes.

> Luckily, there is a better way to handle this problem. In this lesson, you’ll learn about arrays, which are data structures that can store multiple values in a single variable. They are very useful for organizing and manipulating large amounts of data. You’ll also learn about loops, control structures that allow you to execute a block of code repeatedly. Loops are very handy for performing the same operation on each element of an array. Finally, you’ll be introduced to Test-Driven Development (TDD), the practice of writing tests for your code before you create it.

Reviewer 2: ```suggestion
Luckily, there is a better way to handle this problem. In this lesson, you’ll learn about arrays, which are data structures that can store multiple values in a single variable. Arrays are very useful for organizing and manipulating large amounts of data. You’ll also learn about loops, which are control structures that allow you to execute a block of code repeatedly. Loops are very handy for performing the same operation on each element of an array. Finally, you’ll be introduced to Test-Driven Development (TDD), which is the practice of writing tests for your code before you write the code itself.
```

Wonderful job Cake! Considering learners are very new, I reverted back to being a bit more descriptive and repetitive as well. I think being extra clear what is meant with these new terms is crucial.

Author: I don't have strong feelings about either, so I'll follow your feedback.

> Luckily, there is a better way to handle this problem. In this lesson, you’ll learn about arrays, which are data structures that can store multiple values in a single variable. Arrays are very useful for organizing and manipulating large amounts of data. You’ll also learn about loops, which are control structures that allow you to execute a block of code repeatedly. Loops are very handy for performing the same operation on each element of an array. Finally, you’ll also be introduced to a practice of writing tests for your code before you write the code itself, using a practice called Test-Driven Development (TDD).

Reviewer: [Proofreading] This paragraph is off to a great start. It just needed a bit of rewording to make it clear. :) 
```suggestion
Luckily, there is a better way to handle this problem. In this lesson, you’ll learn about arrays, which are data structures that can store multiple values in a single variable. They are very useful for organizing and manipulating large amounts of data. You’ll also learn about loops, control structures that allow you to execute a block of code repeatedly. Loops are very handy for performing the same operation on each element of an array. Finally, you’ll be introduced to Test-Driven Development (TDD). It's where you create tests for your code before writing it.
```

Author: I generally like those changes; the repetition of 'array' was done purposefully, but I can see it might not be necessary and maybe counterproductive. As for the last phrase, I would like to clarify that it's a _practice_ of writing tests.

Reviewer: > I generally like those changes; the repetition of 'array' was done purposefully, but I can see it might not be necessary and maybe counterproductive. As for the last phrase, I would like to clarify that it's a _practice_ of writing tests.

Ok

Author: Thanks for your help!

Reviewer 2: Thanks for providing feedback Christine. 

I have to agree with Cake here that the repetition of array is a good idea since it's the first time people hear about it. 

I also left feedback on the last sentence. Another newly introduced concept in the curriculum
